app/src/main/java/com/example/campusguide/Bootstrapper.kt

```diff
@@ -55,10 +66,24 @@ class Bootstrapper constructor(activity: MapsActivity) {
         activity.setSwitchCampusButtonListener(switchCampus)
 
         // Navigation
-        val route = Route(map, activity)
-        activity.setOnNavigateListener(View.OnClickListener{
-            val chooseDirectionOptions = ChooseDirectionOptionsDialogFragment()
-            chooseDirectionOptions.show(activity.supportFragmentManager, "directionsOptions")
+        activity.setOnNavigateListener(View.OnClickListener {
+            val chooseDestinationOptions = ChooseDestinationOptions { destination ->
```

Comment:
In the future we're going to need to start the direction flow from several locations, so it might be advantageous to have this logic in its own class to be reused - but i wouldn't  mind refactoring that later when we need it

Reply:
For sure, I just wanted to make it so you could use any of the popups from anywhere in the app